components/carousel/utils/useCarouselEvents.ts

```diff
@@ -0,0 +1,151 @@
+import type { CarouselNFT } from '@/components/base/types'
+import type { NFTWithMetadata } from '@/composables/useNft'
+import { isBeta, isProduction } from '@/utils/chain'
+import { formatNFT } from '@/utils/carousel'
+
+import latestEvents from '@/queries/subsquid/general/latestEvents.graphql'
+import latestEventsRmrkv2 from '@/queries/subsquid/ksm/latestEvents.graphql'
+
+interface Types {
+  type: 'latestSales' | 'newestList'
+}
+
+const limit = isProduction ? 15 : 8
+
+const nftEventVariables = {
+  latestSales: { interaction_eq: 'BUY' },
+  newestList: { interaction_eq: 'LIST' },
+}
+
+const disableChainsOnBeta = ['snek']
+
+const fetchLatestEvents = (chain, type, where = {}) => {
+  const { $apollo } = useNuxtApp()
+  const query = chain === 'ksm' ? latestEventsRmrkv2 : latestEvents
+
+  return $apollo.query({
+    query,
+    client: chain,
+    variables: {
+      limit: limit * 2, // buffer the limit
```

Comment:
why *2?

Reply:
Updated, previously to get more varieties of collections. Now will fetch again if nfts.length < limit